shared/crypto/input/index.tsx

```diff
@@ -176,17 +189,117 @@ export const FileInput = (props: FileProps) => {
   )
 }
 
-export const OperationBanner = React.memo((props: OperationBannerProps) => {
-  const color = props.type === 'error' ? 'red' : props.type === 'warning' ? 'yellow' : 'grey'
+export const Input = (props: InputProps) => {
+  const {fileDroppedCounter, operation} = props
+  const dispatch = Container.useDispatch()
+
+  // Store
+  const input = Container.useSelector(state => state.crypto[operation].input.stringValue())
+  const inputType = Container.useSelector(state => state.crypto[operation].inputType)
+
+  // State
+  const [inputValue, setInputValue] = React.useState(input)
+
+  // Actions
+  const onSetInput = React.useCallback(
+    (type: Types.InputTypes, newValue: string) => {
+      dispatch(CryptoGen.createSetInput({operation, type, value: new HiddenString(newValue)}))
+    },
+    [dispatch, operation]
+  )
+  const onClearInput = React.useCallback(() => {
+    dispatch(CryptoGen.createClearInput({operation}))
+  }, [dispatch, operation])
+
+  // Clear the local input state when a user has dragged and dropped a file into the operation
+  // If the input is not cleared then dropping a file, then clearing the file will show old text input
+  const prevCounter = Container.usePrevious(fileDroppedCounter)
+  React.useEffect(() => {
+    if (prevCounter && fileDroppedCounter > prevCounter) {
+      setInputValue('')
+    }
+  }, [fileDroppedCounter, prevCounter])
+
+  return inputType === 'file' ? (
+    <FileInput
+      operation={operation}
+      path={input}
+      onClearFiles={() => {
+        setInputValue('')
+        onClearInput()
+      }}
+    />
+  ) : (
+    <TextInput
+      operation={operation}
+      value={inputValue}
+      onSetFile={path => {
+        onSetInput('file', path)
+      }}
+      onChangeText={text => {
+        setInputValue(text)
+        onSetInput('text', text)
+      }}
+    />
+  )
+}
+
+export const DragAndDrop = (props: DragAndDropProps) => {
+  const {prompt, children, operation, onClearInput} = props
+  const dispatch = Container.useDispatch()
+
+  // Actions
+  const onAttach = React.useCallback(
+    (localPaths: Array<string>) => {
+      const path = localPaths[0]
+      onClearInput()
```

Comment:
is this necessary

Reply:
i see, actually i think this plumbing should change. the input is in the store, so it can just clear it out in the reducer. just clear it in setInput

Reply:
@chrisnojima clearing the input resets the options as well, which I think warrants it being its own action and flow.